src/metabase/lib/native.cljc

```diff
@@ -214,7 +214,7 @@
      (assert-native-query! stage)
      (let [valid-tags (keys existing-tags)]
        (assoc stage :template-tags
-              (m/deep-merge existing-tags (select-keys tags valid-tags)))))))
+              (merge existing-tags (select-keys tags valid-tags)))))))
```

Comment:
`deep-merge` causes issues as we cannot remove properties of existing template tags at all. There is a new test for it.